tests/snap_test.go

```diff
@@ -0,0 +1,201 @@
+package tests
+
+import (
+	"context"
+	"log"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/canonical/matter-snap-testing/utils"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	// enviroment variables
+	specificGpioChip = "GPIO_CHIP"
+	specifGpioLine   = "GPIO_LINE"
+)
+
+var start = time.Now()
+
+const snapName = "matter-pi-gpio-commander"
+const chipToolSnap = "chip-tool"
+
+func TestMain(m *testing.M) {
+	teardown, err := setup() // THIS NEED TO receive "t" but this function doen't have access to it TODO: fix this latter
+	if err != nil {
+		log.Fatalf("Failed to setup tests: %s", err)
+	}
+
+	code := m.Run()
+	teardown()
+
+	os.Exit(code)
+}
+
+const sedCommand = `sed -i 's|/sys/devices/platform/axi/\*.pcie/\*.gpio/gpiochip4/dev|/sys/devices/platform/axi/*.pcie/*.gpio/gpiochip4/dev\n      - /sys/devices/platform/gpio-mockup.*/gpiochip*/dev|' squashfs-root/meta/snap.yaml`
```

Comment:
Instead of replacing (and replicating the target string twice), it is simpler to find and insert after it.

Reply:
Done, changed to: 
```go
const insertMockGPIOAuth = `sed -i '/\/sys\/devices\/platform\/axi\/\*.pcie\/\*.gpio\/gpiochip4\/dev/a \      - /sys/devices/platform/gpio-mockup.*/gpiochip*/dev' squashfs-root/meta/snap.yaml`
```